sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3017,6 +3019,7 @@ object functions {
    * @group datetime_funcs
    * @since 1.5.0
    */
+  @deprecated("This function is deprecated and will be removed in future versions.", "3.0.0")
```

Comment:
BTW, seems like we should deprecate in Python side as well (see https://github.com/apache/spark/commit/d9798c834f3fed060cfd18a8d38c398cb2efcc82 for examples), and R side as well (see https://github.com/apache/spark/commit/0025a8397f8723011917239fe47518457d4d6860#diff-d97f9adc2dcac0703568c799ff106987R352 for instance)

Reply:
@HyukjinKwon Should I do something else besides of adding the functions to `_functions_deprecated`: https://github.com/apache/spark/pull/24195/commits/828da8b2cf6eb873481ea69b69c33d8f7108c784 . Python style checker is not happy with that in the last run.